kinesis/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/sinks/KinesisSink.scala

```diff
@@ -350,82 +394,61 @@ class KinesisSink private (
   private def putToSqs(sqs: SqsClientAndName, batch: List[Event]): Future[Unit] =
     Future {
       log.info(s"Writing ${batch.size} messages to SQS queue: ${sqs.sqsBufferName}")
-      val sqsBatchEntries = batch.map(addUuids).map(toSqsBatchEntry)
       val MaxSqsBatchSize = 10
-      val MaxRetries = 10
-      sqsBatchEntries.grouped(MaxSqsBatchSize).foreach { batchEntryGroup =>
-        sendToSqsWithRetry(sqs, batchEntryGroup, MaxRetries).transform {
-          case failure @ Failure(ex) =>
-            log.info(s"Sending to sqs failed with exception: $ex")
-            failure
-          case notSent @ Success(None) =>
-            log.info(s"Sending to sqs didn't succeed after $MaxRetries retries. Dropping events.")
-            notSent
-          case s @ Success(Some(_)) => s
+      batch
+        .map(toSqsBatchEntry)
+        .grouped(MaxSqsBatchSize)
+        .foreach { batchEntryGroup =>
+          sendToSqs(sqs, batchEntryGroup).transform {
+            case failure @ Failure(ex) =>
+              log.info(s"Sending to sqs failed with exception: $ex")
+              failure
+            case s @ Success(_) => s
+          }
         }
-      }
     }
 
-  /**
-   *  UUIDs are generated for retry purpose. When batch fails, we want to retry (limited number of times)
-   *   only those entries that failed within the batch.
-   */
-  private val addUuids: (Event) => EventWithId =
-    event => EventWithId(event, java.util.UUID.randomUUID())
-
-  private val toSqsBatchEntry: (EventWithId) => (UUID, SendMessageBatchRequestEntry) = {
-    eventWithId =>
-      val b64EncodedMsg = encode(eventWithId.event.msg)
-      val batchReqEntry = new SendMessageBatchRequestEntry(UUID.randomUUID.toString, b64EncodedMsg)
-        .withMessageAttributes(
-          Map(
-            "kinesisKey" ->
-              new MessageAttributeValue()
-                .withDataType("String")
-                .withStringValue(eventWithId.event.key)
-          ).asJava
-        )
-        .withId(eventWithId.id.toString)
-      (eventWithId.id, batchReqEntry)
+  private val toSqsBatchEntry: Event => SendMessageBatchRequestEntry = { event =>
+    val b64EncodedMsg = encode(event.msg)
+    val batchReqEntry = new SendMessageBatchRequestEntry(UUID.randomUUID.toString, b64EncodedMsg)
```

Comment:
Do we still need the UUID if we're not doing 'manual' retries?

Reply:
We don't need this UUID, but there are only two constructors for `SendMessageBatchRequestEntry`, one is parameterless, the other has both id and body, so we set some id and don't use it afterwards.

Reply:
Cool, maybe worth noting that in a comment?